airgun/browser.py

```diff
@@ -49,59 +65,316 @@ class BrowserPlugin(DefaultPlugin):
         '''
 
 
-def browser(browser_name=None, webdriver_name=None):
-    """Creates a webdriver browser.
+class SeleniumBrowserFactory(object):
+    """
 
-    :param browser_name: one of selenium, saucelabs, docker
-    :param webdriver_name: one of firefox, chrome, edge, ie, phantomjs
+    Usage:
```

Comment:
please provide general idea for the factory and why we need it. Put it before `usage` section

Reply:
Yep, i should really start documenting everything, since we're on track and have enough time. Thanks for pointing.